### PR TITLE
Upgrade cores[nginx-1.15.1]

### PR DESCRIPTION
**Pull request by wangfakang**
Delaye to merge that need to update test cases.

### CODE REVIEW

File: src/http/ngx_http_upstream.c [hunk 1]
```diff
@@ -1395,15 +1582,11 @@ ngx_http_upstream_connect(ngx_http_request_t *r, ngx_http_upstream_t *u)
...
-    u->state->response_msec = tp->msec;
-    u->state->header_sec = (time_t) NGX_ERROR;
+    u->state->response_time = ngx_current_msec;
+    u->state->connect_time = (ngx_msec_t) -1;
+    u->state->header_time = (ngx_msec_t) -1;
 
     rc = ngx_event_connect_peer(&u->peer);
-    if (rc == NGX_YIELD) {
-        return;
-    }
```
Comment (chobits) [inline src/http/ngx_http_upstream.c]: delete by mistake?

**Comment by chobits** [comment]
* reviewing diff with nginx-1.15.x: `src/core/`, checked

**Comment by chobits** [comment]
* reviewing diff with nginx-1.15.x:  checked
 * src/core/
 * src/event/  
 * src/mail/   
 * src/misc/ 
 * src os/

**Comment by chobits** [comment]
Note that stream module is not merged in this pull request.

File: src/http/ngx_http_upstream.c (hunk 1 above)
Comment (wangfakang) [reply src/http/ngx_http_upstream.c]: Yea thx,  This PR fix it: https://github.com/alibaba/tengine/pull/1151/commits/10e0d3b19e0aa30f92f0a35ce4d0da90319d7bf7

File: modules/ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.c [hunk 2]
```diff
@@ -603,8 +603,8 @@ ngx_http_upstream_session_sticky_get_peer(ngx_peer_connection_t *pc, void *data)
 
             ctx->sid.len = server[i].sid.len;
             ctx->sid.data = server[i].sid.data;
-
-            sspd->rrp.current = i;
+            /* TODO support */
+            /* sspd->rrp.current = i; */
```
Comment (chobits) [inline modules/ngx_http_upstream_session_sticky_module/ngx_http_upstream_session_sticky_module.c]: reviewing

File: src/core/ngx_cycle.h [hunk 3]
```diff
@@ -141,24 +136,18 @@ ngx_int_t ngx_signal_process(ngx_cycle_t *cycle, char *sig);
...
-#endif
+void ngx_set_shutdown_timer(ngx_cycle_t *cycle);
 
 
 extern volatile ngx_cycle_t  *ngx_cycle;
 extern ngx_array_t            ngx_old_cycles;
 extern ngx_module_t           ngx_core_module;
 extern ngx_uint_t             ngx_test_config;
-extern ngx_uint_t             ngx_show_modules;
-extern ngx_uint_t             ngx_show_directives;
```
Comment (chobits) [inline src/core/ngx_cycle.h]: tag: reviewing

* binary command deleted: ngx_show_modules
* binary command deleted: ngx_show_directives

File: src/core/ngx_log.h [hunk 4]
```diff
@@ -29,15 +29,15 @@
 #define NGX_LOG_DEBUG_EVENT       0x080
 #define NGX_LOG_DEBUG_HTTP        0x100
 #define NGX_LOG_DEBUG_MAIL        0x200
-#define NGX_LOG_DEBUG_MYSQL       0x400
```
Comment (chobits) [inline src/core/ngx_log.h]: tag: reviewing

* maybe make building of following module  failed:

```
$ grep NGX_LOG_DEBUG_MYSQL -R src/
src/mysql/ngx_mysql.c:    ngx_log_debug8(NGX_LOG_DEBUG_MYSQL, rev->log, 0,
```

File: src/http/ngx_http_core_module.c [hunk 5]
```diff
@@ -266,13 +260,6 @@ static ngx_command_t  ngx_http_core_commands[] = {
       offsetof(ngx_http_core_srv_conf_t, large_client_header_buffers),
       NULL },
 
-    { ngx_string("optimize_server_names"),
-      NGX_HTTP_MAIN_CONF|NGX_HTTP_SRV_CONF|NGX_CONF_FLAG,
-      ngx_conf_set_flag_slot,
-      NGX_HTTP_LOC_CONF_OFFSET,
-      offsetof(ngx_http_core_loc_conf_t, server_name_in_redirect),
-      &ngx_conf_deprecated_optimize_server_names },
-
```
Comment (chobits) [inline src/http/ngx_http_core_module.c]: tag: reviewing

* old feature is deleted

File: src/http/ngx_http_core_module.c [hunk 6]
```diff
@@ -553,13 +556,6 @@ static ngx_command_t  ngx_http_core_commands[] = {
       offsetof(ngx_http_core_loc_conf_t, satisfy),
       &ngx_http_core_satisfy },
 
-    { ngx_string("satisfy_any"),
-      NGX_HTTP_MAIN_CONF|NGX_HTTP_SRV_CONF|NGX_HTTP_LOC_CONF|NGX_CONF_FLAG,
-      ngx_conf_set_flag_slot,
-      NGX_HTTP_LOC_CONF_OFFSET,
-      offsetof(ngx_http_core_loc_conf_t, satisfy),
-      &ngx_conf_deprecated_satisfy_any },
-
```
Comment (chobits) [inline src/http/ngx_http_core_module.c]: tag: reviewing
* old feature is deleted.

Comment (chobits) [reply src/http/ngx_http_core_module.c]: ACK: it's nginx feature

File: src/core/ngx_log.h (hunk 4 above)
Comment (chobits) [reply src/core/ngx_log.h]: * should delete all mysql source:

```
$ grep ngx_mysql.c -rl .
./src/mysql/ngx_http_mysql_test.c
./src/mysql/ngx_mysql.c
./src/mysql/config
./src/mysql/ngx_mysql.h
```

* nginx commit:

```
commit 5afd74bed6b9a93b800bf0b5cc8b4e82990e1e9e
Author: Ruslan Ermilov <ru@nginx.com>
Date:   Thu Mar 31 22:33:26 2016 +0300

    Removed the prototype mysql module.
```

Comment (chobits) [reply src/core/ngx_log.h]: fixed in https://github.com/wangfakang/tengine/pull/8

**Comment by chobits** [comment]
* reviewing diff with nginx-1.15.x:  checked
    * src/core/
    * src/event/  
    * src/mail/   
    * src/misc/ 
    * src/os/
    * src/http/
* reviewing diff with last commit of tengine: checked

**Comment by chobits** [comment]
first check

```

Test Summary Report
-------------------
tests/nginx-tests/nginx-tests/memcached_fake.t           (Wstat: 256 Tests: 5 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
tests/nginx-tests/nginx-tests/auth_request_set.t         (Wstat: 1280 Tests: 8 Failed: 5)
  Failed tests:  1-5
  Non-zero exit status: 5
tests/nginx-tests/nginx-tests/auth_request.t             (Wstat: 1792 Tests: 21 Failed: 7)
  Failed tests:  10-12, 14-17
  Non-zero exit status: 7
tests/nginx-tests/nginx-tests/limit_conn_complex.t       (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: Bad plan.  You planned 6 tests but ran 0.
tests/nginx-tests/nginx-tests/gunzip_ssi.t               (Wstat: 512 Tests: 6 Failed: 2)
  Failed tests:  2, 4
  Non-zero exit status: 2
tests/nginx-tests/nginx-tests/gunzip.t                   (Wstat: 256 Tests: 15 Failed: 1)
  Failed test:  7
  Non-zero exit status: 1
tests/nginx-tests/nginx-tests/xslt_params.t              (Wstat: 0 Tests: 6 Failed: 0)
  TODO passed:   2
tests/nginx-tests/nginx-tests/userid.t                   (Wstat: 256 Tests: 35 Failed: 1)
  Failed test:  23
  Non-zero exit status: 1
Files=49, Tests=672, 70 wallclock secs ( 0.27 usr  0.11 sys + 14.81 cusr  3.80 csys = 18.99 CPU)
Result: FAIL
```
```

**Comment by wangfakang** [comment]
checked  tests/nginx-tests/nginx-tests/*.t  success.

TEST_NGINX_BINARY=/home/fakang.wfk/github/tengine/objs/nginx  prove -v -I tests/nginx-tests/nginx-tests/lib/   ./tests/nginx-tests/nginx-tests/

```
All tests successful.

Test Summary Report
-------------------
tests/nginx-tests/nginx-tests/proxy_ssi_body.t                   (Wstat: 0 Tests: 3 Failed: 0)
  TODO passed:   1
tests/nginx-tests/nginx-tests/slice.t                            (Wstat: 0 Tests: 78 Failed: 0)
  TODO passed:   58-59
Files=218, Tests=1876, 122 wallclock secs ( 1.31 usr  0.35 sys + 37.73 cusr  9.54 csys = 48.93 CPU)
Result: PASS
```

**Comment by chobits** [comment]
commit checked:

```
sync nginx-1.15.1 for chunked_filter module.
sync nginx-1.15.1 for proxy module.
sync nginx-1.15.1 for range_filter module.
sync nginx-1.15.1 for ssi_filter module.
sync nginx-1.15.1 for upstream_keepalive module.
```

**Pull request merged**